<?xml version="1.0"?>
<launch>
    <arg name="robot" default="$(optenv ROBOT !!NO_ROBOT_SET!!)"/>
    <arg name="object_perception_target_frame" default="/base_link"/>
    <arg name="base_cmd_topic" default="/cmd_vel_safe"/>
    <arg name="use_mokeup_pose" default="false"/>
    <arg name="domain" default="btt_domain" />
    <arg name="base_path" default="$(find mir_knowledge)/common/pddl/$(arg domain)" />
    <arg name="domain_path" default="$(arg base_path)/domain.pddl" />
    <!--arg name="domain_path" default="$(arg base_path)/domain_with_perception_cost.pddl" /-->
    <arg name="official_refbox" default="cfh1" />
    <arg name="refbox" default="true" />
    <arg name="logging" default="false" />

    <!-- moveit! -->
    <include file="$(find mir_moveit_youbot)/$(arg robot)/move_group.launch">
       <arg name="planner_pipeline" value="interpolation" />
    </include>
    <param name="/move_group/trajectory_execution/allowed_execution_duration_scaling" type="double" value="1.5"/>

    <include file="$(find mir_moveit_client)/ros/launch/moveit_client.launch" />

    <!-- whole body control which also includes pregrasp planner
       it automatically selects wbc pipeline/pregrasp planner pipeline.(see wbc pickup object server for help-->
    <include file="$(find mir_pick_object)/ros/launch/demo_youbot_whole_body_control.launch" >
      <arg name="static_transform_frame" value="$(arg object_perception_target_frame)" />
      <arg name="use_mokeup_pose" value="$(arg use_mokeup_pose)" />
      <arg name="pose_in" value="/mcr_perception/object_selector/output/object_pose" />
      <arg name="cmd_vel_out" value="$(arg base_cmd_topic)" />
    </include>
    
    <!-- dynamic reconfigure from code, receive string and params and does the request NOT TESTED -->
    <include file="$(find mcr_dynamic_reconfigure_client)/ros/launch/dynamic_reconfigure_client.launch">
        <arg name="config_file" value="$(find mir_yb_actions)/ros/config/yb_manipulation_params.yaml" />
    </include>
    
    <!-- wbc pick object action server -->
    <group if="$(arg use_mokeup_pose)">
        <node pkg="mir_pick_object" type="calibrate_pick_server.py" name="calibrate_pick_server"
            output="screen" args="mokeup"/>
    </group>
    <group unless="$(arg use_mokeup_pose)">
        <node pkg="mir_pick_object" type="calibrate_pick_server.py" name="calibrate_pick_server"
            output="screen"/>
    </group>
    <include file="$(find mir_pregrasp_planning)/ros/launch/pose_mock_up_gui_youbot.launch"/>

</launch>
